almond/Dockerfile

```diff
@@ -5,50 +5,77 @@ FROM ${BUILD_FROM}
 ENV \
     LANG="en_US.utf8" \
     THINGENGINE_HOME="/data/almond-server" \
-    THINGENGINE_HOST_BASED_AUTHENTICATION="local-ip"
+    THINGENGINE_HAS_REVERSE_PROXY=true \
+    THINGENGINE_HOST_BASED_AUTHENTICATION=insecure
 
 WORKDIR /opt/almond
 ARG ALMOND_VERSION
 
+RUN adduser --disabled-password almond && \
+   chown almond:almond /opt/almond
+WORKDIR /opt/almond
+
 RUN \
     set -x \
-    && apk add --no-cache \
-        nodejs \
-        npm \
-        yarn \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        build-essential \
+        libcanberra-dev \
+        libcanberra0 \
+        libpulse-dev \
+        libatlas-base-dev \
+        libatlas3-base \
+        pkg-config \
+        ca-certificates \
+        git \
+        gnupg \
         nginx \
+        python-dev \
+        software-properties-common \
         unzip \
-        blas \
         sqlite \
         coreutils \
-    && apk add --no-cache --virtual .build-dependencies \
-        build-base \
-        blas-dev \
-        git \
-        linux-headers \
-        sqlite-dev \
-        pkgconf \
-        python3 \
+        pulseaudio \
+        pulseaudio-utils \
+        gettext \
+        zip \
+        sudo \
+        wget \
+        gstreamer1.0-plugins-base-apps \
+        gstreamer1.0-plugins-good \
+        gstreamer1.0-pulseaudio \
     \
-    && npm config set unsafe-perm \
-    && ln -s /usr/bin/python3 /usr/bin/python \
+    && curl -sS https://deb.nodesource.com/gpgkey/nodesource.gpg.key \
+        | apt-key add - \
+    && echo "deb https://deb.nodesource.com/node_12.x buster main" \
+        > /etc/apt/sources.list.d/nodesource.list \
     \
-    && git clone -b "${ALMOND_VERSION}" --depth 1 \
-        "https://github.com/stanford-oval/almond-server" . \
-    && rm -fr .git \
-    && yarn \
+    && apt-get update \
+    && apt-get install -y --no-install-recommends \
+        nodejs \
     \
-    && yarn global add modclean \
-    && modclean \
-        --path /opt/almond \
-        --no-progress \
-        --keep-empty \
-        --run \
-    && yarn global remove modclean \
+    && npm config set unsafe-perm \
     \
-    && yarn cache clean \
-    && apk del --no-cache \
-        .build-dependencies \
+    && sudo -u almond git clone -b "${ALMOND_VERSION}" --depth 1 \
+        "https://github.com/stanford-oval/almond-server" . && \
+    sudo -u almond npm ci && \
```

Comment:
Sudo?

Reply:
Yes. npm is very picky about running as root, and will generally break. So the source is downloaded and compiled as the regular user instead of root. We could use a USER directive but that would split the image into multiple layers and probably make it larger.

Reply:
The linter fails at this as well. There is no reason to use sudo here.

Reply:
> npm is very picky about running as root, and will generally break.

That is odd, we have quite a lot of add-ons that need NPM that don't show that problem. Any specific reason why this became a problem in this case?